index.html

```diff
@@ -1127,73 +1676,262 @@ <h2>
           </li>
           <li>If the <a>push message</a> contains a payload:
             <ol>
-              <li>Decrypt the <a>push message</a> payload using the private key from the key pair
+              <li>Decrypt the <a>push message</a>'s payload using the private key from the key pair
               associated with |subscription| and the process described in [[RFC8291]]. Set |bytes|
               to the resulting [=/byte sequence=].
               </li>
-              <li>If the <a>push message</a> payload could not be decrypted for any reason:
+              <li>
+                <p>
+                  If the <a>push message</a> payload could not be decrypted for any reason, then
+                  [=acknowledge a push message|acknowledge=] the <a>push message</a> and abort
+                  these steps.
+                </p>
+                <p class="note">
+                  A `push` event is not fired for a <a>push message</a> that was not successfully
+                  decrypted using the key pair associated with the <a>push subscription</a>.
+                </p>
+              </li>
+            </ol>
+          </li>
+          <li>
+            <p>
+              If |bytes| is non-null:
+            </p>
+            <ol>
+              <li>
+                <p>
+                  Let |baseURL| be |registration|'s [=service worker registration/scope URL=].
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |origin| be |baseURL|'s [=url/origin=].
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |fallbackTimestamp| be [=current coarsened wall time=].
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |declarativeResult| be the result of running the [=/declarative push message
+                  parser=] given |bytes|, |origin|, |baseURL|, and |fallbackTimestamp|.
+                </p>
+              </li>
+              <li>
+                <p>
+                  If |declarativeResult| is not failure:
```

Comment:
It's sorta surprising that this implicitly parses all push payloads as declarative push and then falls back, I wonder an explicit option on subscription would make more sense.

Reply:
That would not allow for transitioning existing subscriptions. Being fully backwards compatible is part of the design goals, see the original issue.